### DISCUSSION TITLE
About UWP/WinUI2 and WinAppSdk/WinUI3 Delivery and Runtime Modes

### DISCUSSION THREAD

**Discussion by softworkz** [ffmpeginteropx/FFmpegInteropX#392] [General]
FFmpegInteropX is one of the very few **native code** packages which are aiming to support both, UWP and WinAppSdk applications. A few others are offered by Microsoft but beyond that, it's getting hard to find any others.
That's probably the reason why it's hard to get the required information and understanding from existing documentation, which covers specific parts only, spread over many different places, and sometimes can only be found in GitHub comments from MSFT people.

It took me quite some time get the bigger picture together and still there are some "holes", mainly because I didn't research those parts that I didn't need to know. Let's maintain this post as a work-in-progress and a one-stop reference. I'm making a start now and as questions arise and more information is acquired, we can further amend this post.

## Overview Table

<div class="tg-wrap"><table>
<thead>
  <tr>
    <th></th>
    <th>Packaged</th>
    <th>Container</th>
    <th>Identity</th>
    <th>Process SID</th>
    <th>File Access</th>
    <th>C/C++ Runtimes</th>
    <th>Store Allowed</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td colspan="8">UWP</td>
  </tr>
  <tr>
    <td>Classic UWP</td>
    <td>Yes</td>
    <td>UWP</td>
    <td>Yes </td>
    <td>UWP Container</td>
    <td>Restricted</td>
    <td>App-Local</td>
    <td>Yes (All)</td>
  </tr>
  <tr>
    <td>UWP Desktop</td>
    <td>Yes</td>
    <td>?</td>
    <td>Yes </td>
    <td>UWP Container</td>
    <td>Restricted</td>
    <td>App-Local</td>
    <td>Yes (Windows)</td>
  </tr>
  <tr>
    <td colspan="8">WinAppSDK</td>
  </tr>
  <tr>
    <td>Packaged, Containerized</td>
    <td>Yes</td>
    <td>Light</td>
    <td>Yes </td>
    <td>Special SID (1)</td>
    <td>Restricted</td>
    <td>System</td>
    <td>Yes (Windows)</td>
  </tr>
  <tr>
    <td>Packaged, no Container</td>
    <td>Yes</td>
    <td>No</td>
    <td>Yes </td>
    <td>Current User</td>
    <td>?</td>
    <td>System</td>
    <td>Yes (Windows)</td>
  </tr>
  <tr>
    <td>Not Packaged, Containerized</td>
    <td>No</td>
    <td>Light</td>
    <td>No</td>
    <td>Special SID (2)</td>
    <td>?</td>
    <td>System</td>
    <td>No</td>
  </tr>
  <tr>
    <td>Not Packaged, no Container</td>
    <td>No</td>
    <td>No</td>
    <td>No</td>
    <td>Current User</td>
    <td>Full, like user</td>
    <td>System</td>
    <td>No</td>
  </tr>
</tbody>
</table></div>

(to be continured)

**Comment by Minmoose** [comment]
I don't have a text source for this as it was in a Teams call with a MS engineer last year, but I've been told WinAppSDK's "Packaged, no Container" has **Full Unrestricted** access to the file system just like a "Not Packaged, no Container" app. With that said I've never bothered to test it. 🙃
